src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -730,4 +797,62 @@ private function isUninitializedValueError(\Error $e): bool
         return str_starts_with($e->getMessage(), 'Typed property')
             && str_ends_with($e->getMessage(), 'must not be accessed before initialization');
     }
+
+    private function getNestedAttributes(string $class = null)
+    {
+        if (!$this->classMetadataFactory) {
+            throw new LogicException(sprintf('A class metadata factory must be provided in the constructor when setting "%s" to true.', self::FLATTEN_NESTED_ATTRIBUTES));
+        }
+
+        if (null === $class || !$this->classMetadataFactory->hasMetadataFor($class)) {
+            return [];
+        }
+
+        $properties = [];
+        $classMetadata = $this->classMetadataFactory->getMetadataFor($class);
+        foreach ($classMetadata->getAttributesMetadata() as $name => $metadata) {
+            $serializedName = $metadata->getSerializedName();
+            if (null === $serializedName || false === strpos($serializedName, self::FLATTENER)) {
+                continue;
+            }
+            if (isset($properties[$serializedName])) {
+                throw new LogicException(sprintf('Duplicate serialized name: "%s".', $serializedName));
+            }
+            $properties[$serializedName] = strtolower($name);
```

Comment:
Wouldn't it be smarter to create a hashmap of type array<propertyName, serializedName> instead? This will allow using a direct access operation (O(1)) instead of `in_array()` (O(n)) line 395.

Reply:
Yup. That's way better.